migrations/versions/0428_letter_rates_errata.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+Revision ID: 0428_letter_rates_errata
+Revises: 0427_letter_rates_from_1_11_23
+
+"""
+
+revision = "0428_letter_rates_errata"
+down_revision = "0427_letter_rates_from_1_11_23"
+
+import uuid
+from datetime import datetime
+
+from alembic import op
+from sqlalchemy.sql import text
+
+RATE_CHANGE_DATE = datetime(2023, 11, 1, 0, 0)
+
+CORRECTED_NEW_RATES = [
+    # first class
+    (RATE_CHANGE_DATE, 1, 0.82, "first"),
+    (RATE_CHANGE_DATE, 2, 0.86, "first"),
+    (RATE_CHANGE_DATE, 3, 0.90, "first"),
+    (RATE_CHANGE_DATE, 4, 0.96, "first"),
+    (RATE_CHANGE_DATE, 5, 1.00, "first"),
+]
+
+
+def upgrade():
+    conn = op.get_bind()
+    # DROP the incorrect rates
+    conn.execute(
+        text(
+            """
+        DELETE FROM
+            letter_rates
+        WHERE
+            start_date = :rate_change_date
+        AND
+            post_class = 'first'
+    """
+        ),
+        rate_change_date=RATE_CHANGE_DATE,
+    )
+
+    # add correct new rates
+    for crown in [True, False]:
+        for start_date, sheet_count, rate, post_class in CORRECTED_NEW_RATES:
+            id = uuid.uuid4()
+            conn.execute(
+                text(
+                    """
+                INSERT INTO letter_rates (id, start_date, sheet_count, rate, crown, post_class)
+                    VALUES (:id, :start_date, :sheet_count, :rate, :crown, :post_class)
+            """
+                ),
+                id=id,
+                start_date=start_date,
+                sheet_count=sheet_count,
+                rate=rate,
+                crown=crown,
+                post_class=post_class,
+            )
+
+
+def downgrade():
+    INCORRECT_RATES = [
```

Comment:
im now wondering if there's even a need for this downgrade script.

like, i cant see any value to inserting incorrect rates we know have never been used by dvla. if we don't do anything on downgrade (literally just `pass`), and run upgrade again, it should still delete everything and insert the correct rates without duplicates, which is the key thing i think, to keep the db in a consistent state

Reply:
I agree the only need for the downgrade script would be if DVLA did double take and said the rates were right in the end 😅 .

I can remove it if you think that's better? I'm fine either way ☺️

Reply:
hah if they did a double take i think i'd cry

(but even then, we'd want a _new_ upgrade script, rather than downgrading and then removing this script entirely)

yeah i think removing is better, signifies intent. and having `incorrect_rates` gives me the heebie jeebies 😅.

we use `def downgrade(): pass` a bunch in other migrations that just insert data by the looks of things